2996_reflection/reflection.md

```diff
@@ -4930,6 +4919,21 @@ using alias = [:^^S::type:];    // OK, type-only context
 :::
 :::
 
+### [dcl.mptr]{.sref} Pointers to members {-}
+
+Prefer "designates", and disallow pointers to anonymous unions, in paragraph 2.
+
+::: std
+[2]{.pnum} In a declaration `T D` where `D` has the form
+
+```
+$nested-name-specifier$ * $attribute-specifier-seq$@~_opt_~@ $cv-qualifier-seq$@~_opt_~@ D1
+```
+
+and the `$nested-name-specifier$` [denotes]{.rm} [designates]{.addu} a class, and the type of the contained `$declarator-id$` in the declaration `T D1` is "_derived-declarator-type-list_ `T`", the type of the `$declarator-id$` in `D` is "_derived-declarator-type-list_ `$cv-qualifier-seq$` pointer to member of class `$nested-name-specifier$` of type `T`". The optional `$attribute-specifier-seq$` ([dcl.attr.grammar]) appertains to the pointer-to-member. [`T` shall not be an anonymous union.]{.addu}
```

Comment:
Sorry why are we disallowing this? Cause like:
```cpp
struct C { union { int i; } };
```
Given a `c` it is possible to to to the union object with `c.[:the_union:]` and then to splice again to get to the `i` via `c.[:the_union:].[:the_i:]`. I actually can't tell if this is rejecting `&[:the_union:]` or `&[:the_i:]` (or rather, spelling of those types), but I'm not sure why to reject either?

Reply:
Pointers-to-members `MemTy Cls::*` can't be formed when `Cls` is an anonymous union, and there's some existing core issue stuff to clarify that. The only way to spell the type in the first place is with _splice-type-specifiers_, so I'm arguing that we should just make the type ill-formed in the first place.

Reply:
This is still fine:

```cpp
struct Cls {
  union {
    int mem;
  };
};

auto p = &[:^^Cls::mem:];  // type is 'int Cls::*'.
```

https://godbolt.org/z/EKjdhG8q7

Reply:
Okay so 
```cpp
constexpr auto r = parent_of(^^Cls::mem);
static_assert(is_union_type(r)); // yes
using A = int [:r:]::*; // bad
using B = [:r:]; // fine
using C = [:r:]*; // fine
```
?

Reply:
Aren't you rejecting the wrong type here? In `int [:r:]::*`, `T` is `int`.